internal/controller/pgcluster/pgclustercontroller.go

```diff
@@ -122,7 +122,7 @@ func (c *Controller) processNextItem() bool {
 	found, err := kubeapi.Getpgcluster(c.PgclusterClient, &cluster, keyResourceName, keyNamespace)
 	if !found {
 		log.Debugf("cluster add - pgcluster not found, this is invalid")
-		return false
+		return true
```

Comment:
I think we're going to want to call `Forget` on these before returning `true`.  Otherwise I'm thinking the item will be tracked indefinitely, e.g. by the rate limiter.

Reply:
Very true. There were a few other places that needed the same treatment.